File: backends/php-laravel/app/Domain/Notification/ValueObjects/NotificationListVo.php
<?php

/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */

declare(strict_types=1);

namespace App\Domain\Notification\ValueObjects;

use App\Domain\Notification\Entities\Notification;
use App\Support\Traits\Getter;
use App\Support\ValueObjects\AbstractValueObject;

/**
 * 通知一覧ページの結果 ValueObject です。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Domain\Notification\ValueObjects
 */
class NotificationListVo extends AbstractValueObject
{
    use Getter;

    private array $items = [];
    private ?string $next_cursor = null;
}

File: backends/java-spring/src/main/java/com/authorization/domain/notification/valueobjects/NotificationListVo.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.domain.notification.valueobjects;

import com.authorization.domain.notification.entities.Notification;
import com.authorization.support.valueobjects.AbstractValueObject;
import java.util.List;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 通知一覧ValueObjectクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class NotificationListVo extends AbstractValueObject {

    private List<Notification> items = List.of();
    private String nextCursor;
}
